lnwallet/channel.go

```diff
@@ -6043,9 +6088,25 @@ func (lc *LightningChannel) MayAddOutgoingHtlc(amt lnwire.MilliSatoshi) error {
 func (lc *LightningChannel) htlcAddDescriptor(htlc *lnwire.UpdateAddHTLC,
 	openKey *models.CircuitKey) *paymentDescriptor {
 
+	// TODO(roasbeef): can use push amt to simplify logic, not have to
+	// detect if remote party has enough funds for anchor
+	entryType := Add
+
+	customRecords := htlc.CustomRecords.Copy()
+	if lc.channelState.ChanType.HasTapscriptRoot() {
+		entryType = NoopAdd
```

Comment:
this is a nice way to bootstrap things, but I think `HasTapscriptRoot()` is insufficient as the sole criteria here

We could be sending raw btc HTLCs over the channel, or we could be sending HTLCs carrying custom data and also carry a significant amount of btc. In both of the previous cases we definitely don't want to give the amount back to sender upon settlement

From the top of my head: we could also check for this HTLC's custom records, plus the actual amount of the btc delta and then decide whether we're in a `NoopAdd` case

Reply:
Agreed. I think instead we may want to add a new aux interface, or re-use an existing one, to know when we should be using this or not.